server/game/cards/04-MM/LegionsMarch.js

```diff
@@ -0,0 +1,29 @@
+const Card = require('../../Card.js');
+
+class LegionsMarch extends Card {
+    setupCardAbilities(ability) {
+        this.play({
+            effect:
+                'deal 1 damage to each non-Dinosaur creature after using a Dinosaur creature for the remainder of the turn',
+            gameAction: ability.actions.forRemainderOfTurn((context) => ({
+                when: {
+                    onUseCard: (event) => {
+                        return event.player === context.player && event.card.hasTrait('dinosaur');
```

Comment:
onUseCard is a card action, so the event has no `player` property.  I think you want `event.context.player === context.player`